exact.go

```diff
@@ -3,104 +3,224 @@
 // roudoff errors.
 package exact
 
+import (
+	"math/big"
+)
+
 type (
 	// Frac is a fraction
 	Frac struct {
-		P int64 // P is the numerator
-		Q int64 // Q is the denominator
+		Sign bool
+		P    *big.Int // P is the numerator
+		Q    *big.Int // Q is the denominator
 	}
 )
 
+var (
+	zero = big.NewInt(0)
+	one  = big.NewInt(1)
+
+	// DefPrecision is the default precision used when
+	// the function does not specify one.
+	DefPrecision Frac
+)
+
+func init() {
+	var prec, _, _ = big.NewFloat(0).Parse("1.0e100", 10)
+	z := big.NewInt(0)
+	prec.Int(z)
+	DefPrecision = NewFrac2(big.NewInt(1), z, false)
+}
+
+// One is the whole.
+func One() Frac { return NewFrac2(one, one, false) }
+
+// Zero is the empty.
+func Zero() Frac { return NewFrac2(zero, one, false) }
+
+// NewFrac creates a new fraction using p and q as numerator and
+// denominator, respectively. Sign tells if it's a negative fraction.
+func NewFrac(p, q uint64, sign bool) Frac {
+	return Frac{
+		Sign: sign,
+		P:    big.NewInt(0).SetUint64(p),
+		Q:    big.NewInt(0).SetUint64(q),
+	}
+}
+
+// NewFrac2 creates a new fraction in the same way as NewFrac
+// but using big.Int as numerator and denominator.
+func NewFrac2(p, q *big.Int, sign bool) Frac {
+	if q.Cmp(zero) == 0 {
+		panic("division by zero")
+	}
+	return Frac{
+		Sign: sign,
+		P:    p,
+		Q:    q,
+	}
+}
+
+// IsZero tells if f is zero
+func (f Frac) IsZero() bool {
+	return f.P.Cmp(zero) == 0
+}
+
+// add but ignores sign
+func add(a, b Frac) Frac {
```

Comment:
I was used to add fractions with GCD, did not know of this way of doing it, it seems faster and judging by the tests it seems to work =P

Reply:
It is not GCD, it is LCM that is used on add...sorry I was retarded =P

Reply:
in school, we use the GCD just to add and simplify in one shot =)

Reply:
I see, I have a long way to go x_x